pytorch_lightning/profiler/pytorch.py

```diff
@@ -528,3 +430,372 @@ def teardown(self, stage: Optional[str] = None) -> None:
+                will be used.
+
+            filename: If present, filename where the profiler results will be saved instead of printing to stdout.
+                The ``.txt`` extension will be used automatically.
+
+            group_by_input_shapes: Include operator input shapes and group calls by shape.
+
+            emit_nvtx: Context manager that makes every autograd operation emit an NVTX range
+                Run::
+
+                    nvprof --profile-from-start off -o trace_name.prof -- <regular command here>
+
+                To visualize, you can either use::
+
+                    nvvp trace_name.prof
+                    torch.autograd.profiler.load_nvprof(path)
+
+            export_to_chrome: Whether to export the sequence of profiled operators for Chrome.
+                It will generate a ``.json`` file which can be read by Chrome.
+
+            row_limit: Limit the number of rows in a table, ``-1`` is a special value that
+                removes the limit completely.
+
+            sort_by_key: Attribute used to sort entries. By default
+                they are printed in the same order as they were registered.
+                Valid keys include: ``cpu_time``, ``cuda_time``, ``cpu_time_total``,
+                ``cuda_time_total``, ``cpu_memory_usage``, ``cuda_memory_usage``,
+                ``self_cpu_memory_usage``, ``self_cuda_memory_usage``, ``count``.
+
+            record_functions: Set of profiled functions which will create a context manager on.
+                Any other will be pass through.
+
+            record_module_names: Whether to add module names while recording autograd operation.
+
+            profiler_kwargs: Keyword arguments for the PyTorch profiler. This depends on your PyTorch version
+
+        Raises:
+            MisconfigurationException:
+                If arg ``sort_by_key`` is not present in ``AVAILABLE_SORT_KEYS``.
+        """
+        super().__init__(
+            dirpath,
+            filename,
+            group_by_input_shapes,
+            emit_nvtx,
+            export_to_chrome,
+            row_limit,
+            sort_by_key,
+            record_functions,
+            record_module_names,
+            **profiler_kwargs,
+        )
+
+        # calcuate the warmup and active step
+        self._action_step_num: Dict[str, int] = defaultdict(int)
+        self._wait_step: int = wait_step
+        self._warmup_step: int = warmup_step
+        self._active_step: int = active_step
+
+        if _KINETO_AVAILABLE:
+            self._init_kineto_params(profiler_kwargs)
+
+    def _get_profiler_class(self) -> Type:
+        return torch.profiler._KinetoProfile if _KINETO_AVAILABLE else torch.autograd.profiler.profile
+
+    def _init_profiler(self, action_name: str) -> None:
+        self._create_profilers()
+
+        if self._parent_profiler is not None:
+            self._parent_profiler.__enter__()
+
+        if not _KINETO_AVAILABLE or self._emit_nvtx:
```

Comment:
does it mean that if not _KINETO_AVAILABLE or self._emit_nvtx, the profiler is entered at init, so the wait_step/warmup_step/activate_step would not work?

Reply:
you are right, if self._emit_nvtx is True or _KINETO_AVAILABLE not available, the self.profiler will be either autograd profiler or nvtx profiler. 
As the result, the self.stop will not invoke the wait/warmup/active step anymore.
The behavior is consistent with current logic.

Reply:
need add comment at the function argument?